src/plugins/intel_cpu/src/emitters/tpp/x64/jit_eltwise_emitters.cpp

```diff
@@ -0,0 +1,153 @@
+    if (dim0 == dim1) {
+        bcast_flags = {false, false};
+        return dim0;
+    } else if (dim1 == 1) {
+        bcast_flags = {false, true};
+        return dim0;
+    } else if (dim0 == 1) {
+        bcast_flags = {true, false};
+        return dim1;
+    }
+    OPENVINO_THROW("Invalid dimensions passed to get_broadcast_flags");
+    return -1;
+}
+
+UnaryEltwiseTppEmitter::UnaryEltwiseTppEmitter(dnnl::impl::cpu::x64::jit_generator* h,
+                                               dnnl::impl::cpu::x64::cpu_isa_t isa,
+                                               const ov::snippets::lowered::ExpressionPtr& expr) :
+                                               TppEmitter(h, isa, expr) {
+    const auto& subtensor_in0 = get_projected_subtensor(io_port_descriptors[0]);
+
+    const auto N = static_cast<libxsmm_blasint>(*subtensor_in0.rbegin());
+    const auto M = static_cast<libxsmm_blasint>(*++subtensor_in0.rbegin());
+
+    const auto& unary_eltw_tpp = std::dynamic_pointer_cast<tpp::op::UnaryEltwiseTPP>(expr->get_node());
+    OPENVINO_ASSERT(unary_eltw_tpp, "Invalid TPP node type detected in UnaryEltwiseTppEmitter");
+    m_op_type = unary_eltw_tpp->get_op_type();
+    // Note: libxsmm implies column-major layout, so we have to swap M and N here
+    m_shape = libxsmm_create_meltw_unary_shape(N, M,
+                                               io_strides[0], io_strides[1],
+                                               io_dtypes[0], io_dtypes[1],
+                                               dtype_comp);
+    m_compile_flags = LIBXSMM_MELTW_FLAG_UNARY_NONE;
+}
+
+void UnaryEltwiseTppEmitter::execute_kernel(libxsmm_meltwfunction_unary eltwise_kernel, void *in0, void *out0) {
+    libxsmm_meltw_unary_param param;
+    param.op.primary = nullptr;
+    param.in.primary = in0;
+    param.out.primary = out0;
+    eltwise_kernel(&param);
+}
+
+std::set<std::vector<element::Type>> UnaryEltwiseTppEmitter::get_supported_precisions(const std::shared_ptr<ov::Node>& node) {
+    return {{element::f32}};
+}
+
+void UnaryEltwiseTppEmitter::validate_arguments(const std::vector<size_t> &in, const std::vector<size_t> &out) const {
+    OPENVINO_ASSERT(in.size() == 1, "UnaryEltwiseTppEmitter expects 1 input registers, got " + std::to_string(in.size()));
+    OPENVINO_ASSERT(out.size() == 1, "UnaryEltwiseTppEmitter expects 1 output register, got " + std::to_string(out.size()));
+}
+
+ReduceTppEmitter::ReduceTppEmitter(dnnl::impl::cpu::x64::jit_generator* h,
+                                   dnnl::impl::cpu::x64::cpu_isa_t isa,
+                                   const ov::snippets::lowered::ExpressionPtr& expr)  :
+                                   UnaryEltwiseTppEmitter(h, isa, expr) {
+    m_compile_flags = LIBXSMM_MELTW_FLAG_UNARY_REDUCE_ROWS;
+    // No need to set ldo for reduce, it is always assumed = 1 inside the kernel
+    // m_shape.ldo = 1;
```

Comment:
I would like to keep it if you don't mind. It explains why we can pass invalid out leading dimensions in case of Reduce operations. If we remove it, the `ldo` choice for Reduce might be confusing.

Reply:
Thanks for the explanation! I agree, let's leave the comment